api/pkg/s3/objectput.go

```diff
@@ -15,92 +15,188 @@
 package s3
 
 import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
 	"net/http"
 	"strconv"
 	"strings"
-	"time"
 
 	"github.com/emicklei/go-restful"
-	log "github.com/sirupsen/logrus"
+        log "github.com/sirupsen/logrus"
 	"github.com/opensds/multi-cloud/api/pkg/common"
-	c "github.com/opensds/multi-cloud/api/pkg/context"
-	"github.com/opensds/multi-cloud/api/pkg/s3/datastore"
-	"github.com/opensds/multi-cloud/api/pkg/utils/constants"
-	. "github.com/opensds/multi-cloud/s3/pkg/exception"
-	"github.com/opensds/multi-cloud/s3/pkg/utils"
+	"github.com/opensds/multi-cloud/s3/error"
 	"github.com/opensds/multi-cloud/s3/proto"
+	pb "github.com/opensds/multi-cloud/s3/proto"
 )
 
+var ChunkSize int = 2048
+
 //ObjectPut -
 func (s *APIService) ObjectPut(request *restful.Request, response *restful.Response) {
```

Comment:
does this API match the existing yaml API spec ? if not, please update and share new spec.

Reply:
@rhsakarpos  I have update spec in path "openapi-spec/swagger.yaml"

Reply:
ok